students/Kate_Koehn/Lesson09/jpgdiscover.py

```diff
@@ -0,0 +1,29 @@
+"""
+Recursively go through directories looking for pngs
+"""
+
+import os
+
+
+def jpg_discovery(directory, png_paths=None):
+    """
+    Search directory recursively for all png files and add to dict
+    """
+    files = os.listdir(directory)
+    for file in files:
+        file_path = os.path.join(directory, file)
+        if os.path.isdir(file_path):
+            jpegdiscovery(file_path)
+        elif file.endswith(".png"):
+            if directory in file_dict.keys():
+                file_dict[directory].append(file)
+            else:
+                file_dict[directory] = [file]
+
```

Comment:
What's up with the dictionary? If you want to use a dictionary, you should be able to add it as another keyword argument, initialize when it is None, and not have to declare it outside of the function, which is not very user friendly.